### ISSUE TITLE
bug: `index()` does not work for web and mobile targets, but says it does in docs

### ISSUE THREAD

**Issue by kilacoda** [keymanapp/keyman#10964]
### Describe the bug

Hi,

I'm trying to define rules for a touch keyboard for Devanagari, and when I try to compile it I get the following error:

![image](https://github.com/keymanapp/keyman/assets/14952412/882386ec-a557-49cf-b4ee-884abecf49b2)

However, the documentation page for `index()` says otherwise:

![image](https://github.com/keymanapp/keyman/assets/14952412/2e06aeec-7d72-4443-8ad0-8144ff1eb3fc)

The `.kmn` file I'm trying to compile:

[hindi.txt](https://github.com/keymanapp/keyman/files/14530293/hindi.txt)

**Keyman Developer Version:** 17.0.283-beta

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [X] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.283-beta

### Operating system

Windows 10 22H2

### Language name

Hindi

**Comment by mcdurdin** [comment]
Yes, we should tweak the documentation for this to clarify that `index()` is only available in the output part of the rule for mobile / web, for v17. For v18, we hope to improve this.
